src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -37,6 +37,22 @@ public static bool IsReservedKeyword(SyntaxKind kind)
             return kind >= SyntaxKind.BoolKeyword && kind <= SyntaxKind.ImplicitKeyword;
         }
 
+        public static bool IsControlKeyword(SyntaxKind kind)
+        {
+            return kind == SyntaxKind.YieldKeyword
+                || kind == SyntaxKind.WhenKeyword
+                || kind == SyntaxKind.InKeyword
```

Comment:
why is 'in' a control keyword?

Reply:
I was thinking it would be appropriate when it is part of a foreach statement. I will add a comment, but I am also open to being talked out of this position. =)